src/clients/repos.js

```diff
@@ -0,0 +1,122 @@
+import trim from 'lodash/trim';
+import gitHub from '../services/gitHub';
+import generatePreview from '../util/generatePreview';
+import performWithRetries from '../util/performWithRetries';
+
+const COMMIT_MESSAGE = 'Created using Popcode: https://popcode.org';
+const MASTER_BRANCH = 'master';
+
+// Sometimes there are race conditions with creating multiple files in separate
+// API requests
+function repoPerformWithRetries(perform) {
+  return performWithRetries(perform, ['Not Found']);
+}
+
+async function createHtmlFile(github, project, userName, repoName) {
+  if (trim(project.sources.html)) {
+    return github.getRepo(userName, repoName).
+      writeFile(
+        MASTER_BRANCH,
+        'raw_index.html',
+        project.sources.html,
+        COMMIT_MESSAGE,
+        {},
+      );
+  }
+  return Promise.resolve();
+}
+
+async function createCssFile(github, project, userName, repoName) {
+  if (trim(project.sources.css)) {
+    return github.getRepo(userName, repoName).
+      writeFile(
+        MASTER_BRANCH,
+        'styles.css',
+        project.sources.css,
+        COMMIT_MESSAGE,
+        {},
+      );
+  }
+  return Promise.resolve();
+}
+
+async function createJsFile(github, project, userName, repoName) {
+  if (trim(project.sources.javascript)) {
+    return github.getRepo(userName, repoName).writeFile(
+      MASTER_BRANCH, 'script.js', project.sources.js, COMMIT_MESSAGE, {});
+  }
+  return Promise.resolve();
+}
+
+async function createPreviewFile(github, project, userName, repoName) {
+  if (trim(project.sources.html) ||
+    trim(project.sources.css) ||
+    trim(project.sources.javascript)) {
+    const preview = generatePreview(
+      project,
+      {
+        targetBaseTop: false,
+        propagateErrorsToParent: false,
+        breakLoops: false,
+        nonBlockingAlertsAndPrompts: false,
+      },
+    );
+    return github.getRepo(userName, repoName).
+      writeFile(MASTER_BRANCH, 'index.html', preview, COMMIT_MESSAGE, {});
```

Comment:
(A side effect of this is we now create the repo with `auto_init` set to true - this touches an empty README. Otherwise the API doesn't like you creating and pushing a branch in a repo with nothing in it.